### PR TITLE
Task/update android sdk(4.2.0)/sdk 957

### PR DESCRIPTION
**Pull request by william-ct**
*Update ct-core deps to v4.2.0
*NPE crash fix
*FCM token refresh issue fix

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## CHANGE LOG
 
+### Version 1.2.3 (July 19, 2021)
+* Supports [CleverTap Android SDK v4.2.0](https://github.com/CleverTap/clevertap-android-sdk/releases/tag/core-v4.2.0)
+* Adds fix for NPE [Github Issue](https://github.com/CleverTap/clevertap-flutter/issues/58)
```
Comment (darshanclevertap) [inline CHANGELOG.md:5]: Mention Adds fix for `recordScreen` NPE as well. Issue #54

**Comment by darshanclevertap** [comment]
@william-ct You need to updated the SDK version in `pubspec.yaml` file as well

**Review by darshanclevertap — APPROVED** [review]
LGTM 👍🏾

**Comment by Aditi3** [comment]
@darshanclevertap @william-ct version upgrade in iOS podspec looks missing!

**Pull request merged**